### PR TITLE
split ecs out ready for a lambda scanner

### PR DESCRIPTION
**Pull request by progers-tech**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (blackmamo) [inline .vscode/settings.json:1]: Please don't commit IDE specific files, add to your .git/info/exclude

File: infrastructure/docker_image/Dockerfile.template [hunk 2]
```diff
@@ -1,6 +1,7 @@
 FROM alpine:latest
 
 # Install nmap
+
```
Comment (blackmamo) [inline infrastructure/docker_image/Dockerfile.template:4]: Non semantic change, and no improvement, please remove the noise

File: infrastructure/infrastructure.tf [hunk 3]
```diff
@@ -79,8 +79,29 @@ module "elastic_resources" {
   ssm_source_stage = "${local.ssm_source_stage}"
 }
 
+module "nmap_task_shared" {
```
Comment (blackmamo) [inline infrastructure/infrastructure.tf]: I really don't like this shared name. We are not sharing this module. nmap_task_resources?

File: infrastructure/infrastructure.tf [hunk 4]
```diff
@@ -79,8 +79,29 @@ module "elastic_resources" {
...
+  subscribe_elastic_to_notifier = true
+  account_id                    = "${var.account_id}"
+  ssm_source_stage              = "${local.ssm_source_stage}"
+  transient_workspace           = "${!contains(var.known_deployment_stages, terraform.workspace)}"
+  task_role_name                = "${module.nmap_task.task_role_name}"
+}
+
 module "nmap_task" {
-  // two slashes are intentional: https://www.terraform.io/docs/modules/sources.html#modules-in-package-sub-directories
+  // two slashes are intentional: https://www.terraform.io/docs/modules/sources.html#modules-in-package-sub-directories  # source = "github.com/ministryofjustice/securityanalytics-taskexecution//infrastructure/ecs_task"
```
Comment (blackmamo) [inline infrastructure/infrastructure.tf]: Don't need that additional commented out line here

File: infrastructure/infrastructure.tf (hunk 3 above)
Comment (blackmamo) [inline infrastructure/infrastructure.tf]: We don't want to have to import two modules here. We only want to import the ecs_task. The ecs_task should itself use the simpler module

**Review by blackmamo — CHANGES_REQUESTED** [review]
This has not been done in a way to make the user interface simple. The user has to import 2 modules where previously they only needed to import 1

**Comment by progers-tech** [comment]
not needed

**Pull request closed without merging**